libs/vm-modules/src/math/tensor/tensor_estimator.cpp

```diff
@@ -93,65 +95,153 @@ ChargeAmount TensorEstimator::SetAtFour(uint64_t /*idx1*/, uint64_t /*idx2*/, ui
 
 ChargeAmount TensorEstimator::Fill(DataType const & /*value*/)
 {
-  return ComputeChargeFromTensorSize();
+  SizeType padded_size = fetch::math::Tensor<DataType>::PaddedSizeFromShape(tensor_.shape());
+  SizeType size        = fetch::math::Tensor<DataType>::SizeFromShape(tensor_.shape());
+
+  return static_cast<ChargeAmount>(FILL_PADDED_SIZE_COEF() * padded_size + FILL_SIZE_COEF() * size +
+                                   FILL_CONST_COEF()) *
+         COMPUTE_CHARGE_COST;
 }
 
 ChargeAmount TensorEstimator::FillRandom()
 {
-  return ComputeChargeFromTensorSize();
+  SizeType padded_size = fetch::math::Tensor<DataType>::PaddedSizeFromShape(tensor_.shape());
+  SizeType size        = fetch::math::Tensor<DataType>::SizeFromShape(tensor_.shape());
+
+  return static_cast<ChargeAmount>(FILL_RANDOM_PADDED_SIZE_COEF() * padded_size +
+                                   FILL_RANDOM_SIZE_COEF() * size + FILL_RANDOM_CONST_COEF()) *
+         COMPUTE_CHARGE_COST;
 }
 
 ChargeAmount TensorEstimator::Min()
 {
-  return ComputeChargeFromTensorSize();
+  SizeType padded_size = fetch::math::Tensor<DataType>::PaddedSizeFromShape(tensor_.shape());
+  SizeType size        = fetch::math::Tensor<DataType>::SizeFromShape(tensor_.shape());
+
+  return static_cast<ChargeAmount>(MIN_PADDED_SIZE_COEF() * padded_size + MIN_SIZE_COEF() * size +
+                                   MIN_CONST_COEF()) *
+         COMPUTE_CHARGE_COST;
 }
 
 ChargeAmount TensorEstimator::Max()
 {
-  return ComputeChargeFromTensorSize();
+  SizeType padded_size = fetch::math::Tensor<DataType>::PaddedSizeFromShape(tensor_.shape());
+  SizeType size        = fetch::math::Tensor<DataType>::SizeFromShape(tensor_.shape());
+
+  return static_cast<ChargeAmount>(MAX_PADDED_SIZE_COEF() * padded_size + MAX_SIZE_COEF() * size +
+                                   MAX_CONST_COEF()) *
+         COMPUTE_CHARGE_COST;
 }
 
 ChargeAmount TensorEstimator::Squeeze()
 {
-  return ComputeChargeFromTensorSize();
+  auto new_shape = tensor_.shape();
+
+  bool     not_found = true;
+  SizeType cur_dim   = new_shape.size() - 1;
+  while (not_found)
+  {
+    if (new_shape.at(cur_dim) == static_cast<SizeType>(1))
+    {
+      new_shape.erase(new_shape.begin() + static_cast<int32_t>(cur_dim));
+      not_found = false;
+      break;
+    }
+    else
+    {
+      if (cur_dim == 0)
+      {
+        break;
+      }
+      --cur_dim;
+    }
+  }
+
+  if (not_found)
```

Comment:
estimators dont need logic like this; if the behaviour is invalid it'll throw anyway.
Also check with volodymyr that the tensor behaviour for squeeze is going to stay like this or whether it'l being updated (i.e. numpy squeeze i think doesn't throw if there's nothing to squeeze, it just doesn't do anything)